embedchain/config/base_app_config.py

```diff
@@ -18,6 +18,7 @@ def __init__(
         id: Optional[str] = None,
         collect_metrics: bool = True,
         collection_name: Optional[str] = None,
+        cache: Optional[bool] = False,
```

Comment:
May be change this to be a dict instead of a bool so that we can configure the cache settings.

```
"cache": {
    "similarity_threshold": 0.5,
    "max_distance": 1.0
}
```

Reply:
I was thinking of allowing `similarity_threshold` to be configurable for now with range [0,1]. GPTCache uses the following formula to calculate the baseline threshold which is complex and confusing for users - 
```
baseline_threshold = (max_distance - min_distance) * similarity_threshold * cache_factor
```
The defaults are:
max_distance = 4.0, we change it to 1.0 so that similarity_threshold works as the actual threshold
min_distance = 0.0
cache_factor = 1.0

Let me know what you think.

Reply:
Discussed offline. We will add support for this in follow up PR.